llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1921,6 +1917,7 @@ MemoryDepChecker::getDependenceDistanceStrideAndSize(
   if (StrideAPtr && *StrideAPtr < 0) {
     std::swap(Src, Sink);
     std::swap(AInst, BInst);
+    std::swap(ATy, BTy);
```

Comment:
This looks like a bug fix and unrelated to this patch - can you commit this in a separate PR please?